cv_next/providers/supabase-provider.tsx

```diff
@@ -1,50 +1,70 @@
-"use client";
+'use client'
 
-import { createContext, useContext, useEffect, useState } from "react";
-import { createPagesBrowserClient } from "@supabase/auth-helpers-nextjs";
-import { useRouter } from "next/navigation";
+import { createContext, useContext, useEffect, useState } from 'react'
+import { Session, SupabaseClient, createClientComponentClient } from '@supabase/auth-helpers-nextjs'
+import { useRouter } from 'next/navigation'
 
-import type { SupabaseClient } from "@supabase/auth-helpers-nextjs";
+type MaybeSession = Session | null
 
 type SupabaseContext = {
-	supabase: SupabaseClient;
-};
+  supabase: SupabaseClient<any, string>
+  session: MaybeSession
+}
 
-const Context = createContext<SupabaseContext | undefined>(undefined);
+const Context = createContext<SupabaseContext | undefined>(undefined)
 
 export default function SupabaseProvider({
-	children,
+  children,
+  session,
 }: {
-	children: React.ReactNode;
+  children: React.ReactNode
+  session: MaybeSession
 }) {
-	const [supabase] = useState(() => createPagesBrowserClient());
-	const router = useRouter();
-
-	useEffect(() => {
-		const {
-			data: { subscription },
-		} = supabase.auth.onAuthStateChange(() => {
-			router.refresh();
-		});
-
-		return () => {
-			subscription.unsubscribe();
-		};
-	}, [router, supabase]);
-
-	return (
-		<Context.Provider value={{ supabase }}>
-			<>{children}</>
-		</Context.Provider>
-	);
+  const supabase = createClientComponentClient()
+  const router = useRouter()
+
+  useEffect(() => {
+    const {
+      data: { subscription },
+    } = supabase.auth.onAuthStateChange((_, _session) => {
+      if (_session?.access_token !== session?.access_token) {
+        router.refresh()
+      }
+    })
+
+    return () => {
+      subscription.unsubscribe()
+    }
+  }, [router, supabase, session])
+
+  return (
+    <Context.Provider value={{ supabase, session }}>
+      <>{children}</>
+    </Context.Provider>
+  )
+}
+
+export const useSupabase = <
```

Comment:
I think these hooks (`useSupabase`,  `useSession`) should sit in a /hooks folder, not in the provider file